presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java

```diff
@@ -74,7 +82,40 @@ public void removeAgent(UUID id)
         }
     }
 
-    //TODO: Add method to generate ID
+    public synchronized void refreshNodes()
+    {
+        List<ServiceDescriptor> services = serviceSelector.selectAllServices();
+
+        ImmutableSet.Builder<Agent> setBuilder = ImmutableSet.builder();
+
+        for (ServiceDescriptor service : services) {
+            Map<String, String> properties = service.getProperties();
+            // TODO: Make agents start without an ID, and provide one on first discovery
+            UUID id = service.getId();
+            boolean isCoordinator = Boolean.parseBoolean(properties.get("presto-coordinator"));
```

Comment:
What do I not need within the loop? Every value assigned in the loop is used in the loop and replaced in the next iteration.